benches/search_comparison.rs

```diff
@@ -285,3 +240,64 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+fn test_sequence<T>(duplicates: bool, max: usize) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    let mut index = 0;
+    std::iter::from_fn(move || {
+        let item = if duplicates {
+            let item = std::cmp::min(index, max);
+            item % 256
+        } else {
+            std::cmp::min(index * 2, max)
+        };
+        index += 1;
+
+        Some(T::try_from(item).unwrap())
+    })
+}
+
+/// Generate pseudorandom sequence of numbers
+///
+/// Without duplicates:
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u8>(16, 0, false, u8::MAX).take(16).collect(),
+///     vec![30, 4, 18, 8, 6, 12, 26, 16, 14, 20, 2, 24, 22, 28, 10, 0]
+/// )
+/// ```
+///
+/// With duplicates:
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u32>(16, 0, true, u32::MAX).take(16).collect(),
+///     vec![15, 14, 9, 8, 5, 10, 0, 4, 5, 0, 4, 6, 5, 14, 12, 2]
+/// );
+/// ```
+fn pseudorandom_iter<T>(
+    size: usize,
+    mut seed: usize,
+    duplicates: bool,
+    max: usize,
+) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    std::iter::from_fn(move || {
+        // LCG constants from https://en.wikipedia.org/wiki/Numerical_Recipes.
+        seed = seed.wrapping_mul(1664525).wrapping_add(1013904223);
+
+        if duplicates {
```

Comment:
There is very simple way how to repeat items in the iterator without imperative code:

```rust
iter.flat_map(|v| std::iter::repeat(v).take(duplicates))
```

This way we can isolate random number generation and duplicating them.

Reply:
Okay, makes sense, what number would `duplicates` (perhaps `duplicate_factor` would be better term in that case?) be equal to? Static `2`? or something like `size // 4`?

Reply:
I suggest to leave it equal to 16 (as it is right now).

https://github.com/jonhoo/ordsearch/blob/2b576b9b04d1142813a3d0e91ba9ba4eb01f49f8/benches/search_comparison.rs#L238

We can change it in the future if we need to.

Reply:
Okay, but that would make tests of duplicates with size <= 16 contain only a single unique number, is that what you want?

Reply:
Yes, this is how it works right now basically. Again this "anomaly" only appears in tests like `Search (with duplicates) u8/ordsearch/8` where there is quite narrow integer type and small collection size.

It is not the best solution. But I believe it should be addressed separately.